CI upstream-dev issue TypeError: Cannot interpret '<StringDtype(na_value=nan)>' as a data type

# traceback
```python
safe_s1/reader.py:731: in get_noise_azi_raw
    ds = xr.concat(tmp, pd.Index(pols, name="pol"))
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../micromamba/envs/tests/lib/python3.12/site-packages/xarray/structure/concat.py:324: in concat
    return _dataset_concat(
../../../micromamba/envs/tests/lib/python3.12/site-packages/xarray/structure/concat.py:852: in _dataset_concat
    index_vars = index.create_variables()
                 ^^^^^^^^^^^^^^^^^^^^^^^^
../../../micromamba/envs/tests/lib/python3.12/site-packages/xarray/core/indexes.py:799: in create_variables
    data = PandasIndexingAdapter(self.index, dtype=self.coord_dtype)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'PandasIndexingAdapter' object has no attribute '_dtype'") raised in repr()] PandasIndexingAdapter object at 0x7f75fcfafb50>
array = Index(['VV', 'VH'], dtype='str', name='pol')
dtype = <StringDtype(na_value=nan)>

    def __init__(
        self,
        array: pd.Index,
        dtype: DTypeLike | pd.api.extensions.ExtensionDtype | None = None,
    ):
        from xarray.core.indexes import safe_cast_to_index
    
        self.array = safe_cast_to_index(array)
    
        if dtype is None:
            if is_allowed_extension_array(array):
                cast(pd.api.extensions.ExtensionDtype, array.dtype)
                self._dtype = array.dtype
            else:
                self._dtype = get_valid_numpy_dtype(array)
        elif is_allowed_extension_array_dtype(dtype):
            self._dtype = cast(pd.api.extensions.ExtensionDtype, dtype)
        else:
>           self._dtype = np.dtype(cast(DTypeLike, dtype))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           TypeError: Cannot interpret '<StringDtype(na_value=nan)>' as a data type

../../../micromamba/envs/tests/lib/python3.12/site-packages/xarray/core/indexing.py:1920: TypeError
----------------- generated report log file: pytest-log.jsonl ------------------
=========================== short test summary info ============================
FAILED test/test_s1reader.py::test_reader - TypeError: Cannot interpret '<StringDtype(na_value=nan)>' as a data type
============================== 1 failed in 5.51s ===============================

```

--- a/safe_s1/__init__.py
+++ b/safe_s1/__init__.py
@@ -1,6 +1,6 @@
 import traceback
 
-from safe_s1.reader import Sentinel1Reader
+from safe_s1.reader import Sentinel1Reader  # noqa: F401
 
 try:
     from importlib import metadata

--- a/safe_s1/reader.py
+++ b/safe_s1/reader.py
@@ -728,7 +728,7 @@ class Sentinel1Reader:
 
                 noise_lut_azi_raw_ds[vari].attrs["description"] = hihi
             tmp.append(noise_lut_azi_raw_ds)
-        ds = xr.concat(tmp, pd.Index(pols, name="pol"))
+        ds = xr.concat(tmp, pd.Index(pols, name="pol", dtype=object))
         ds.attrs["history"] = "\n".join(history)
         return ds
 
@@ -767,7 +767,6 @@ class Sentinel1Reader:
                     )
                 )
             tmp.append(noise_lut_range_raw_ds)
-        # ds = xr.concat(tmp, pd.Index(pols, name="pol"))
         ds = xr.concat(tmp, pd.Index(pols, name="pol", dtype=object))
         ds.attrs["history"] = "\n".join(history)
         return ds

--- a/safe_s1/reader.py
+++ b/safe_s1/reader.py
@@ -767,7 +767,8 @@ class Sentinel1Reader:
                     )
                 )
             tmp.append(noise_lut_range_raw_ds)
-        ds = xr.concat(tmp, pd.Index(pols, name="pol"))
+        # ds = xr.concat(tmp, pd.Index(pols, name="pol"))
+        ds = xr.concat(tmp, pd.Index(pols, name="pol", dtype=object))
         ds.attrs["history"] = "\n".join(history)
         return ds
 

--- a/safe_s1/sentinel1_xml_mappings.py
+++ b/safe_s1/sentinel1_xml_mappings.py
@@ -784,7 +784,7 @@ def df_files(annotation_files, measurement_files, noise_files, calibration_files
     # we want them uniques, and compatibles with gdal sentinel driver (ie 'WV_012')
     pols_count = len(set(pols))
     subds_count = len(annotation_files) // pols_count
-    dsid_count = len(set(dsid))
+    # dsid_count = len(set(dsid))
     # if dsid_count != subds_count:
     if "WV" in dsid[0]:
         dsid_rad = dsid[0][:-1]  # WV

